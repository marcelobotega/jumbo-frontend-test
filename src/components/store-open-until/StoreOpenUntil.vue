<template>
  <div class="store-open-until form-group md:w-1/2 lg:w-1/4 w-full">
    <label class="form-label">Open until:</label>
    <select
      name="open-until"
      autocomplete="off"
      class="form-control"
      v-model="selectValue"
      @change="selectedTime"
    >
      <option value="">Select time</option>
      <option
        v-for="openUntil in openUntilList"
        :key="openUntil"
        :value="openUntil"
      >
        {{ openUntil }}
      </option>
    </select>
  </div>
</template>

<script lang="ts">
import { Component, Emit, Vue } from "vue-property-decorator";
import storeService from "@/services/store.service";

@Component
export default class StoreOpenUntil extends Vue {
  selectValue = "";
  openUntilList = storeService.getStoresOpenUntilValues();

  @Emit()
  selectedTime(): string {
    return this.selectValue;
  }
}
</script>
