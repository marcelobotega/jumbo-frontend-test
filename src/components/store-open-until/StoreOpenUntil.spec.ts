import { shallowMount, Wrapper } from "@vue/test-utils";
import StoreOpenUntil from "@/components/store-open-until/StoreOpenUntil.vue";
import Vue from "vue";

describe("StoreOpenUntil.vue", () => {
  let wrapper: Wrapper<StoreOpenUntil>;
  let openUntilSelect: Wrapper<Vue>;

  beforeEach(() => {
    wrapper = shallowMount(StoreOpenUntil);
    openUntilSelect = wrapper.find('select[name="open-until"]');
  });

  it("should create", () => {
    expect(wrapper.vm.$data.selectValue).toEqual("");
    expect(wrapper.vm.$data.openUntilList).toEqual([
      "18:00",
      "19:00",
      "20:00",
      "20:30",
      "21:00",
      "21:30",
      "22:00",
      "23:00",
    ]);
    expect(wrapper).toMatchSnapshot();
  });

  it("should emit value when change select", async () => {
    await wrapper.setData({ selectValue: "19:00" });
    await openUntilSelect.trigger("change");
    await wrapper.vm.$nextTick();
    expect(wrapper).toMatchSnapshot();

    await wrapper.setData({ selectValue: "23:00" });
    await openUntilSelect.trigger("change");
    await wrapper.vm.$nextTick();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.emitted()).toEqual({
      "selected-time": [
        ["19:00", expect.any(Object)],
        ["23:00", expect.any(Object)],
      ],
    });
  });
});
