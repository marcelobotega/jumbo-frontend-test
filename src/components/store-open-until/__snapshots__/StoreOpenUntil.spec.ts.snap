// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreOpenUntil.vue should create 1`] = `
<div class="store-open-until form-group md:w-1/2 lg:w-1/4 w-full"><label class="form-label">Open until:</label> <select name="open-until" autocomplete="off" class="form-control">
    <option value="">Select time</option>
    <option value="18:00">
      18:00
    </option>
    <option value="19:00">
      19:00
    </option>
    <option value="20:00">
      20:00
    </option>
    <option value="20:30">
      20:30
    </option>
    <option value="21:00">
      21:00
    </option>
    <option value="21:30">
      21:30
    </option>
    <option value="22:00">
      22:00
    </option>
    <option value="23:00">
      23:00
    </option>
  </select></div>
`;

exports[`StoreOpenUntil.vue should emit value when change select 1`] = `
<div class="store-open-until form-group md:w-1/2 lg:w-1/4 w-full"><label class="form-label">Open until:</label> <select name="open-until" autocomplete="off" class="form-control">
    <option value="">Select time</option>
    <option value="18:00">
      18:00
    </option>
    <option value="19:00">
      19:00
    </option>
    <option value="20:00">
      20:00
    </option>
    <option value="20:30">
      20:30
    </option>
    <option value="21:00">
      21:00
    </option>
    <option value="21:30">
      21:30
    </option>
    <option value="22:00">
      22:00
    </option>
    <option value="23:00">
      23:00
    </option>
  </select></div>
`;

exports[`StoreOpenUntil.vue should emit value when change select 2`] = `
<div class="store-open-until form-group md:w-1/2 lg:w-1/4 w-full"><label class="form-label">Open until:</label> <select name="open-until" autocomplete="off" class="form-control">
    <option value="">Select time</option>
    <option value="18:00">
      18:00
    </option>
    <option value="19:00">
      19:00
    </option>
    <option value="20:00">
      20:00
    </option>
    <option value="20:30">
      20:30
    </option>
    <option value="21:00">
      21:00
    </option>
    <option value="21:30">
      21:30
    </option>
    <option value="22:00">
      22:00
    </option>
    <option value="23:00">
      23:00
    </option>
  </select></div>
`;
